### DISCUSSION TITLE
One instance of PrismaClient + SvelteKit

### DISCUSSION THREAD

**Discussion by chris-carrington** [prisma/prisma#17386] [Q&A]
Thank you for the Sveltekit example! Is the prisma.ts file in the example the recommended approach to ensure we only create one instance of prisma client please?!

https://github.com/prisma/prisma-examples/blob/latest/typescript/rest-sveltekit/src/lib/prisma.ts

Where I learned about this: https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client

Thank you!

**Comment by nurul3101** [comment]
Hi @chris-carrington 👋 

Yes, the approach in the svelte-kit example is ideal and would ensure creating of a single PrismaClient instance.
For reference, we have a [help article](https://www.prisma.io/docs/guides/database/troubleshooting-orm/help-articles/nextjs-prisma-client-dev-practices) that is focused on best practices of Instantiating a Prisma Client Instance in Next.js project, this is useful while developing as due to hot reloading there can be several instances of Prisma Client being created. The approach could also be helpful for Svelte Kit if hot reloading is used in Svelte Kit.

**Reply by chris-carrington** [reply]
Thank you @nurul3101  for the swift reply and additional information! You've taught me so much in this post! Do you see any disadvantages if I accomplish one instance like this please?!

```
/**
 * Prisma Recommendation: "Your application should generally only create one instance of PrismaClient": https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client
 * Typescript global extend: https://stackoverflow.com/a/68328575 + https://bobbyhadz.com/blog/typescript-element-implicitly-has-any-type-typeof-globalthis
 */
import { PrismaClient } from '@prisma/client'

declare global {
  var prisma: PrismaClient
}

if (!global.prisma) global.prisma = new PrismaClient()

export default global.prisma
```

**Reply by nurul3101** [reply]
Thank you so much for your kind words! 🙂 
No, I don't see any issue in instantiating one Prisma client like this.

**Reply by chris-carrington** [reply]
@nurul3101 You Da Man!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
